alpine/latest/Dockerfile

```diff
@@ -0,0 +1,86 @@
+###############################################################
+#
+# Copyright (C) 2019 Olliver Schinagl <oliver@schinagl.nl>
+#
+# SPDX-License-Identifier: MIT
+
+# For Alpine, latest is actually the latest stable
+# hadolint ignore=DL3007
+###############################################################
+FROM registry.hub.docker.com/library/alpine:latest AS builder
+
+###############################################################
+# set build args
+###############################################################
+ARG CURL_RELEASE_TAG=latest
+ARG CURL_GIT_REPO=https://github.com/curl/curl.git
+ARG CURL_CONFIGURE_OPTION
+ARG LABEL_VERSION=1.0.0
+ARG LABEL_NAME=curl
+ARG LABEL_DESC=curl
+
+###############################################################
+# build curl
+###############################################################
+# We want the latest stable version from the repo
+# hadolint ignore=DL3018
+RUN \
+    apk add --no-cache \
+       autoconf \
+       automake \
+       build-base \
+       curl-dev \
+       groff \
+       libtool \
+    && \
+    rm -rf "/var/cache/apk/"*
+
+RUN mkdir /src
+COPY "curl"  "/src/curl"
+WORKDIR "/src/curl"
+
+RUN ./buildconf && \
+    autoreconf -vif && \
+    ./configure ${CURL_CONFIGURE_OPTION} \
+    && make \
+    && make DESTDIR="/alpine/" install
+
+###############################################################
+# For Alpine, latest is actually the latest stable
+# hadolint ignore=DL3007
+FROM registry.hub.docker.com/library/alpine:latest
+
+###############################################################
+# define docker labels
+###############################################################
+LABEL Maintainer="Olliver Schinagl <oliver@schinagl.nl>"
+LABEL Name="curl"
+LABEL Version="${LABEL_VERSION}"
+LABEL docker.cmd="docker run -it curl/curl-alpine http://www.google.com"
+
+###############################################################
+# remove curl
+###############################################################
+RUN \
+    apk add --no-cache \
+        curl \
```

Comment:
So this step:

- Installs the (apk version of curl)'s runtime dependencies
- Then systematically deletes all the files in the APK package?

I don't think that gets us where we want to be - the development dependencies aren't installed up above and therefore we're not linking against them.

Not sure about this step tbh. Preference would be downloading and compiling curl statically where possible.